src/main/java/org/testng/internal/Parameters.java

```diff
@@ -190,6 +232,21 @@ private static void checkParameterTypes(String methodName,
       }
     }
 
+    if ( (parameterNames.length ==0) && (totalLength != 0) ) {
+      //parameterNames is usually populated via the @Parameters annotation, so we would need to
+      //apply our logic only when @Parameters annotation is not involved.
+      String errPrefix = "Cannot inject " + methodAnnotation + " annotated Method [" + methodName + "] with "
+            + Arrays.toString(parameterTypes);
+      if (mapping.containsKey(methodAnnotation)) {
```

Comment:
Replace `String methodAnnotation` by `Annotation methodAnnotation`, it will be stronger.